### DISCUSSION TITLE
How do Hydraulics come out of RAS?

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#308] [Show and tell]
**Unsteady Flow Hydraulics**
Unsteady flow hydraulics come out of RAS as either saved grids, or pXX.hdf files.
- HDF
      Are saved in the project folder with all the other intermediate, inputs, data storage files. They are not the only .hdf files, but are always pXX.hdf. Often RAS projects have extra runs that were done for sensitivity you may not be interested in using, so it's probably going to be best that the hydraulic engineer pulls out all the important results files into a separate directory to hand off to the FDA lead. 
![image](https://user-images.githubusercontent.com/64556723/159788400-315db2fb-a1db-44b2-885f-af0c7710a198.png)
![image](https://user-images.githubusercontent.com/64556723/159789312-efaf0c89-d262-4927-9796-d32336e9a07c.png)

- Saved Grids
     Are saved in subdirectories of the project folder, named with the plan short name of the plan file. Grids typically are saved with a .vrt whether they are in multiple pieces or not. As discussed elsewhere, RAS gridded results will be broken up in the same manner the terrain they're based on, so there may be many tifs and 1 vrt, or a single tif.  If multiple maps are made for a given plan, there will both be saved in the same subdirectory
![image](https://user-images.githubusercontent.com/64556723/159788185-045cd98a-b5a6-4718-a39a-97bdc904b21d.png)
![image](https://user-images.githubusercontent.com/64556723/159788333-c4b6a6df-dd98-49b1-84bd-7703e42a617a.png)

- Saved Points
     This is my personal preferred mode of sharing hydraulics the more I think and play with it. RAS can append WSE or Depth to a point shapefile. The Data is saved in the same manner as grids. In the same file structure that is. The benefit here is it's a fraction of the size. The downside is you can't ever move those structure points without regenerating these shapefiles.
![image](https://user-images.githubusercontent.com/64556723/160011992-ea014372-6e24-4dbc-a221-693b59883e3e.png)

     RAS just appends whatever type of map you have to the attribute table with a field name of Map Value. It's The picture below is WSE. The ground elevation field is coming from the NSI shapefile we're using, not the terrain value from the RAS Model. 
![image](https://user-images.githubusercontent.com/64556723/160012878-97e3fe2a-144e-4bc9-8000-d0b65c4cfc06.png)

**Steady Flow***
Steady flow results are saved in the same manner as uncertain, they do exist in a different place in the .hdf file, however. Also interesting, is that all 8 profiles will typically be saved in the same, single .hdf file, rather than 8 separate like the unsteady one's above. All saved raster output also exists in a single subdiretory
![image](https://user-images.githubusercontent.com/64556723/159789076-345f4837-aa49-4706-9864-bd1868a7357d.png)
![image](https://user-images.githubusercontent.com/64556723/159792597-ac805398-144c-4cba-a1f5-c5afdc3c4724.png)

**Comment by rnugent3** [comment]
@Brennan1994 thank you for walking us through the format of the data - this is really helpful. 

Would it be reasonable to ask the HEC-RAS team to standardize the way that the data is exported?

**Reply by Brennan1994** [reply]
I see we have a meeting set now to discuss. I think their organization makes a lot of sense for a hydraulic modeler to understand it. It's just not necessarily convenient for us. I would lean toward just working with it. We will discuss more our meeting though. I intend to write something up in one of the other threads to flesh out what I think the best way to solve this problem is.

**Reply by rnugent3** [reply]
Thank you, sir.
